pyro/infer/mcmc/api.py

```diff
@@ -408,8 +408,12 @@ def model(data):
         # If transforms is not explicitly provided, infer automatically using
         # model args, kwargs.
         if self.transforms is None:
+            # Try to initialize kernel.transforms using kernel.setup().
+            if getattr(self.kernel, "transforms", None) is None:
+                warmup_steps = 0
+                self.kernel.setup(warmup_steps, *args, **kwargs)
             # Use `kernel.transforms` when available
-            if hasattr(self.kernel, 'transforms') and self.kernel.transforms is not None:
+            if getattr(self.kernel, "transforms", None) is not None:
```

Comment:
This is required because manually calling `initialize_model()` below would ignore `init_strategy` and attempt to sample from `ImproperUniform`. Instead we let `kernel.setup()` call `initialize_model()` with proper arguments.